tests/test_core.py

```diff
@@ -1061,6 +1061,134 @@ def test_select_P_ABBA_val_inf():
     npt.assert_almost_equal(ref, comp)
 
 
+def test_merge_topk_PI():
+    # `assume_unique = True`
+    n = 50
+    for k in range(1, 6):
+        PA = np.random.rand(n * k).reshape(n, k)
+        PA[:, :] = np.sort(PA, axis=1)  # sorting each row separately
+
+        PB = np.random.rand(n * k).reshape(n, k)
+        col_idx = np.random.randint(0, k, size=n)
+        for i in range(n):  # creating ties between values of PA and PB
+            PB[i, col_idx[i]] = np.random.choice(PA[i], size=1, replace=False)
+        PB[:, :] = np.sort(PB, axis=1)  # sorting each row separately
+
+        IA = np.arange(n * k).reshape(n, k)
+        IB = IA + n * k
+
+        ref_P = PA.copy()
+        ref_I = IA.copy()
+
+        comp_P = PA.copy()
+        comp_I = IA.copy()
+
+        naive.merge_topk_PI(ref_P, PB.copy(), ref_I, IB.copy())
+        core._merge_topk_PI(comp_P, PB.copy(), comp_I, IB.copy())
+
+        npt.assert_array_equal(ref_P, comp_P)
+        npt.assert_array_equal(ref_I, comp_I)
+
+    # `assume_unique = False`
```

Comment:
Replace this comment with: 
"# rest case: there are overlapping values between each row of IA, e.g. IA[i] and its corresponding row in IB, i.e. IB[i]"

Reply:
Then I think you should split this unit test up into smaller tests. One called `test_merge_topk_PI_without_overlap` and `test_merge_topk_PI_with_overlap` and then add this comment.